src/debugging/netSdk/NetSdkDebugHelper.ts

```diff
@@ -124,24 +110,46 @@ export class NetSdkDebugHelper extends NetCoreDebugHelper {
         return appOutput;
     }
 
-    private inferDotNetSdkContainerName(debugConfiguration: DockerDebugConfiguration): string {
-        const projFileUri = Uri.file(path.dirname(debugConfiguration.netCore.appProject));
-        return getDefaultContainerName(path.basename(projFileUri.fsPath), "dev");
+    private async getProjectInfo(debugConfiguration: DockerDebugConfiguration): Promise<string[]> {
+        if (this.projectInfo !== undefined && this.projectInfo.length >= 6) {
+            return this.projectInfo;
+        }
+
+        const ridOS = await normalizeOsToRidOs();
+        const ridArchitecture = await normalizeArchitectureToRidArchitecture();
+        const additionalProperties = `/p:ContainerRuntimeIdentifier="${ridOS}-${ridArchitecture}"`;
+
+        let projectInfo: string[];
+        try {
+            projectInfo = await getNetCoreProjectInfo('GetProjectProperties', debugConfiguration.netCore?.appProject, additionalProperties);
+        } catch (error) {
+            await ext.context.workspaceState.update(NetContainerBuildOptionsKey, '');
+            throw error;
+        }
+
+        if (projectInfo.length < 6 || !projectInfo[5]) {
+            await ext.context.workspaceState.update(NetContainerBuildOptionsKey, '');
+            throw new Error(l10n.t("Your current project configuration or .NET SDK version doesn't support SDK Container build. Please choose a compatible project or update .NET SDK."));
+        }
+
+        this.projectInfo = projectInfo;
```

Comment:
Consider normalizing the projectInfo array into something easier to parse in code (an object or similar) to avoid potential regressions if the layout of the array ever changes.

Reply:
Like a dictionary / hashmap?

Reply:
update: i made an interface